sanic/mixins/listeners.py

```diff
@@ -26,12 +26,33 @@ def __init__(self, *args, **kwargs) -> None:
     def _apply_listener(self, listener: FutureListener):
         raise NotImplementedError  # noqa
 
+    @overload
+    def listener(
+        self,
+        listener_or_event: ListenerType[Sanic],
+        event_or_none: str,
+        apply: bool = True,
```

Comment:
Do not need a default here.

Reply:
won't this make the typing invalid? I can't get the type checking to work as intended without the default. Removing them would make them required.

Reply:
Oh, nice. Just found out one can use an ellipsis (`...`). Thanks for teaching me that 😄